### ISSUE TITLE
Piwik update notification email links to the internal Piwik instance IP address but should link to the trusted hostname value instead

### ISSUE THREAD

**Issue by mattab** [matomo-org/matomo#11029]
Report from email:

Update notification uses IP address instead of the (first) valid hostname from the config.

```
Hallo,

Er is een nieuwe versie van Piwik beschikbaar als update

U kunt automatisch updaten naar versie 3.0.0-rc2 of het pakket downloaden en handmatig installeren.
https://192.168.210.10/index.php?module=CoreUpdater&action=newVersionAvailable

Voel je vrij om je ideeën en suggesties te delen met het Piwik team:
http://piwik.org/contact/
```

* 192.168.210.10 should be instead "analytics.example.com" from the trusted_hosts array in the config.ini.php

**Comment by mattab** [comment]
This may be the same issue as: https://github.com/piwik/piwik/issues/6644

**Comment by pestevao** [comment]
I think that they are differente issues.
It only happens to us this in email reports and always resolve to 127.0.0.1 which is not used to configure any trusted_hosts entries.

**Comment by GavL89** [comment]
@tsteur - in relation to the closed issue #14413, no, none of the trusted hostnames are IP addresses.  This is what the general section of our config.ini.pho file looks like

[General]
salt = "xxxxxxxxxx"
trusted_hosts[] = "analytics.ap-southeast-2.domain.com"
proxy_client_headers[] = "HTTP_X_FORWARDED_FOR"
proxy_host_headers[] = "HTTP_X_FORWARDED_HOST"
cors_domains[] = "*"
session_save_handler = dbtable
multi_server_environment=1
noreply_email_address = "noreply@domain.com"

**Comment by tsteur** [comment]
Was also reported here: https://github.com/matomo-org/matomo/issues/14413

I wonder: Maybe `SettingsPiwik::getPiwikUrl` should only store a url/hostname that is configured in the trusted hostnames once Matomo is installed and shouldn't overwrite a URL that isn't set as trusted url? Not sure if that would cause any issues but might be a good idea.

> I've also tried updating the piwikUrl row in the database however it just gets updated by itself to the IP address of one of the servers.

@GavL89 it does look like the proxy header isn't actually forwarded cause the app should use it otherwise.

**Comment by tsteur** [comment]
@GavL89 I reckon we have kind of an idea where the problem is and will likely create a fix for this as part of 3.10 or 3.11
